src/batou_ext/resources/mailpit.nix

```diff
@@ -0,0 +1,23 @@
+{ lib, pkgs, ... }: {
+  systemd.services.mailpit = {
+    wantedBy = [ "multi-user.target" ];
+    serviceConfig = {
+      DynamicUser = true;
+      StateDirectory = "mailpit";
+      WorkingDirectory = "%S/mailpit";
+      ExecStart = "${lib.getExe pkgs.mailpit} -d mailpit.db -s \"[::]:{{component.smtp_port}}\" -l \"[::1]:{{component.ui_port}}\"";
+      Restart = "on-failure";
+    };
+  };
```

Comment:
I think this belongs into a nixos module upstream. Will prepare a PR there when everything else is done.